### PR TITLE
Update the modularity doc and minor changes to the modularity method

### PR DESCRIPTION
**Pull request by vgauthier**
1. Enhance the documentation with references and example. 
2. Add the resolution parameter inside the computation of the modularity.
3. add new tests for the modularity function.

### CODE REVIEW

File: src/community/modularity.jl [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+
+julia> add_edge!(g, 3, 1);
+
+julia> add_edge!(g, 3, 4);
+
+julia> add_edge!(g, 4, 5);
+
+julia> add_edge!(g, 5, 6);
+
+julia> add_edge!(g, 6, 4);
```
[the remark this answers is not in the document]
Comment (vgauthier) [reply src/community/modularity.jl]: I updated the code according to your comments

**Comment by simonschoelly** [comment]
Looks good to me. What is the convention on latex in docstrings for LightGraphs?

**Comment by vgauthier** [comment]
let's me know if there are guide lines to handle  latex in docstrings for LightGraphs, I will make the change accordingly.

**Comment by sbromberger** [comment]
> What is the convention on latex in docstrings for LightGraphs?

It's fine to use LaTeX in docstrings - we do so where it makes sense.

**Pull request merged**